documentation/source/domains.rst

```diff
@@ -67,36 +67,44 @@ Function Examples
 
 Given the following Breathe directives::
 
-   .. doxygenfunction:: testnamespace::NamespacedClassTest::function
-      :path: ../../examples/specific/class/xml
+   .. doxygennamespace:: TestNamespaceFunction
+      :project: cpp_function
```

Comment:
Can you help me understand the purpose of this kind of change? From doxygenfunction to doxgennamespace?

Reply:
Its a problem in referencing the function. To be more concise, the `doxygennamespace` directive does not accept `:members:` option as a list. 

```rst
.. doxgenfunction:: NS::func

reference to func() using :cpp:`NS:func()`
.. error::
    The entity NS is not documented!
    Cannot find reference to NS::func()

But, improperly referencing tends to work better if
we ignore the undocumented namespace: observe :cpp:`func()`.
```
The above is a simplified scenario of what this change fixes

Reply:
Unless we provide the `doxygennamespace` directive with the ability to only document certain members (should be a separate issue), I guess I should reword the example here to better explain why we need to document the namespace of a member instead of only the member.

Reply:
I realize now that I may have gone overboard trying to solve some nit-picky mode warnings. Reverting this change does not trigger any warnings (with nit-picky mode disabled).

I'm putting together a sample project for a new issue about the `enumvalue`/`enumerator` reference not working.